modules/playlists/src/main/java/org/opencastproject/playlists/serialization/JaxbPlaylist.java

```diff
@@ -0,0 +1,191 @@
+import javax.xml.bind.annotation.XmlType;
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+/** 1:1 serialization of a {@link Playlist}. Intended for endpoints. */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "playlist", namespace = "http://playlist.opencastproject.org")
+@XmlRootElement(name = "playlist", namespace = "http://playlist.opencastproject.org")
+public class JaxbPlaylist {
+
+  static class DateAdapter extends XmlAdapter<Long, Date> {
+    /**
+     * {@inheritDoc}
+     *
+     * @see javax.xml.bind.annotation.adapters.XmlAdapter#marshal(java.lang.Object)
+     */
+    @Override
+    public Long marshal(Date v) throws Exception {
+      return v == null ? null : v.getTime();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see javax.xml.bind.annotation.adapters.XmlAdapter#unmarshal(java.lang.Object)
+     */
+    @Override
+    public Date unmarshal(Long v) throws Exception {
+      return v == null ? null : new Date(v);
+    }
+  }
+
+  @XmlAttribute()
+  private String id;
+
+  @XmlElement(name = "organization")
+  private String organization;
+
+  private List<JaxbPlaylistEntry> entries;
+
+  @XmlElement(name = "title")
+  private String title;
+
+  @XmlElement(name = "description")
+  private String description;
+
+  @XmlElement(name = "creator")
+  private String creator;
+
+  @XmlJavaTypeAdapter(JaxbPlaylist.DateAdapter.class)
+  @XmlElement
+  private Date updated;
+
+  private List<JaxbPlaylistAccessControlEntry> accessControlEntries ;
+
+  void beforeMarshal(Marshaller u) {
+    // Explicitly set empty lists to `null`. This is to avoid having an empty list wrongly show up in a JSON
+    // serialization with the value of an empty string
+    if (entries != null && entries.isEmpty()) {
+      entries = null;
+    }
+    if (accessControlEntries != null && accessControlEntries.isEmpty()) {
+      accessControlEntries = null;
+    }
+  }
+
+  /**
+   * Default no-arg constructor needed by JAXB
+   */
+  public JaxbPlaylist() {
+  }
+
+  public JaxbPlaylist(Playlist playlist) {
+    this();
+    this.id = playlist.getId();
+    this.organization = playlist.getOrganization();
+    this.entries = playlist.getEntries()
+        .stream()
+        .map(JaxbPlaylistEntry::new)
+        .collect(Collectors.toList());
+    this.title = playlist.getTitle();
+    this.description = playlist.getDescription();
+    this.creator = playlist.getCreator();
+    this.updated = playlist.getUpdated();
+    this.accessControlEntries = playlist.getAccessControlEntries()
+        .stream()
+        .map(JaxbPlaylistAccessControlEntry::new)
+        .collect(Collectors.toList());
+  }
+
+  public Playlist toPlaylist() {
+    return new Playlist(
+        id,
+        organization,
+        Optional.ofNullable(entries).orElseGet(Collections::emptyList)
+        .stream().map(JaxbPlaylistEntry::toPlaylistEntry).collect(Collectors.toList()),
```

Comment:
Formatting is wonky. Either do it like with `accessControlEntries` below or like this (similar to above):

```suggestion
        Optional.ofNullable(entries)
            .orElseGet(Collections::emptyList)
            .stream()
            .map(JaxbPlaylistEntry::toPlaylistEntry)
            .collect(Collectors.toList()),
```